common/lib/dataset.py

```diff
@@ -377,10 +376,32 @@ def iterate_items(self, processor=None, warn_unmappable=True, map_missing="defau
 							mapped_item.data[missing_field] = mapped_item.data[missing_field].value
 						else:
 							raise ValueError("map_missing must be 'abort', 'default', or a callback.")
-
 			else:
 				mapped_item = original_item
 
+			# Add possible annotations
+			if annotation_labels:
+
+				# Get annotations for this specific post
+				post_annotations = self.get_annotations(item_id=mapped_item.data["id"])
```

Comment:
I get a crash here after adding a multiple-choice annotation field to a dataset:

```
ERROR:webtool:Exception on /results/74a03678e6a669f8db441ac431f31de2/explorer/ [GET]
Traceback (most recent call last):
  File "/Users/stijn/surfdrive/PycharmProjects/4cat/venv/lib/python3.9/site-packages/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/stijn/surfdrive/PycharmProjects/4cat/venv/lib/python3.9/site-packages/flask/app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/stijn/surfdrive/PycharmProjects/4cat/venv/lib/python3.9/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/stijn/surfdrive/PycharmProjects/4cat/venv/lib/python3.9/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/stijn/surfdrive/PycharmProjects/4cat/venv/lib/python3.9/site-packages/flask_limiter/extension.py", line 544, in __inner
    return obj(*a, **k)
  File "/Users/stijn/surfdrive/PycharmProjects/4cat/venv/lib/python3.9/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
  File "/Users/stijn/surfdrive/PycharmProjects/4cat/webtool/lib/helpers.py", line 374, in decorated_view
    return func(*args, **kwargs)
  File "/Users/stijn/surfdrive/PycharmProjects/4cat/webtool/views/views_explorer.py", line 97, in explorer_dataset
    for row in dataset.iterate_items(warn_unmappable=False):
  File "/Users/stijn/surfdrive/PycharmProjects/4cat/common/lib/dataset.py", line 386, in iterate_items
    post_annotations = self.get_annotations(item_id=mapped_item.data["id"])
AttributeError: 'dict' object has no attribute 'data'

```

Reply:
Fixed in [55d81f5](https://github.com/digitalmethodsinitiative/4cat/pull/428/commits/55d81f5217feb6380b6ff014bb67384401833fb1)